packages/trpc/server/routers/viewer/eventTypes/util.ts

```diff
@@ -152,6 +152,40 @@ export function ensureUniqueBookingFields(fields: z.infer<typeof EventTypeUpdate
   }, {} as Record<string, true>);
 }
 
+export function ensureEmailOrPhoneNumberIsPresent(
+  fields: z.infer<typeof EventTypeUpdateInput>["bookingFields"]
+) {
+  if (!fields || fields.length === 0) {
+    return;
+  }
+
+  const attendeePhoneNumberField = fields.find((field) => field.name === "attendeePhoneNumber");
+
+  const emailField = fields.find((field) => field.name === "email");
+
+  if (emailField?.hidden && attendeePhoneNumberField?.hidden) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: `Both Email and Attendee Phone Number cannot be hidden`,
+    });
+  } else if (emailField?.hidden && emailField?.required) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: `Email field cannot be hidden if it is a required field. You can click on Edit to mark it as not required field.`,
+    });
+  } else if (attendeePhoneNumberField?.hidden && attendeePhoneNumberField?.required) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: `Attendee Phone Number field cannot be hidden if it is a required field. You can click on Edit to mark it as not required field.`,
+    });
+  } else if (!emailField?.required && !attendeePhoneNumberField?.required) {
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: `Atleast Email or Attendee Phone Number need to be required field.`,
+    });
+  }
+}
```

Comment:
We don't need `else if` in here since we're throwing and other conditions anyways. 

```suggestion
  if (emailField?.hidden && attendeePhoneNumberField?.hidden) {
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: `Both Email and Attendee Phone Number cannot be hidden`,
    });
  }
  if (emailField?.hidden && emailField?.required) {
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: `Email field cannot be hidden if it is a required field. You can click on Edit to mark it as not required field.`,
    });
  }
  if (attendeePhoneNumberField?.hidden && attendeePhoneNumberField?.required) {
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: `Attendee Phone Number field cannot be hidden if it is a required field. You can click on Edit to mark it as not required field.`,
    });
  }
  if (!emailField?.required && !attendeePhoneNumberField?.required) {
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: `Atleast Email or Attendee Phone Number need to be required field.`,
    });
  }
```